src/hotspot/share/oops/method.cpp

```diff
@@ -407,8 +408,9 @@ void Method::metaspace_pointers_do(MetaspaceClosure* it) {
 
 void Method::remove_unshareable_info() {
   unlink_method();
-  if (AOTCodeCache::is_dumping_adapter() && _adapter != nullptr) {
+  if (CDSConfig::is_dumping_adapters() && _adapter != nullptr) {
     _adapter->remove_unshareable_info();
+    _adapter = nullptr;
```

Comment:
We shouldn't be setting _adapter to null here. This breaks the link between Method and its adapter in the AOT cache.

Reply:
Without nulling it I got issue with referencing dead memory through this field.

Reply:
I think that should not be happening. Can you please create a new issue for this with more details and I can try why is that the case.

Reply:
The number of cached adapters and performance improvement are still here:
```
$ (perf stat -r 100 ./build/product/images/jdk/bin/java -XX:AOTCache=app.aotcache -XX:+UnlockDiagnosticVMOptions -XX:+AOTAdapterCaching -cp hello.jar HelloWorld > /dev/null) 2>&1 | grep elapsed
         0.0247033 +- 0.0000319 seconds time elapsed  ( +-  0.13% )

vs w/o AOT adapters:
         0.0266180 +- 0.0000352 seconds time elapsed  ( +-  0.13% )
```